x-pack/plugins/lens/public/datasources/form_based/to_esql.test.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IndexPattern } from '../../types';
+import { getESQLForLayer } from './to_esql';
+import { createCoreSetupMock } from '@kbn/core-lifecycle-browser-mocks/src/core_setup.mock';
+import { DateHistogramIndexPatternColumn } from '../..';
+
+describe('to_esql', () => {
```

Comment:
We should probably include some more tests with all implemented functions and cases